config/externalname.go

```diff
@@ -3228,3 +3228,36 @@ func eksPodIdentityAssociation() config.ExternalName {
 	}
 	return e
 }
+
+// fullArnTemplate builds a templated string for constructing a terraform id component which is an ARN, which includes
+// the aws partition, service, region, account id, and resource. This is by far the most common form of ARN.
+// e.g. arn:aws:ec2:ap-south-1:123456789012:instance/i-1234567890ab
+func fullArnTemplate(service string, resource string) string {
+	return genericArnTemplate(service, resource, false, false)
+
+}
+
+// regionlessArnTemplate builds a templated string for constructing a terraform id component which is an ARN of a
+// resource which is regionless, but specific to your account id. It includes the partition, service, account id, and
+// resource.
+// e.g. arn:aws:iam::123456789012:role/example
+func regionlessArnTemplate(service string, resource string) string {
+	return genericArnTemplate(service, resource, false, true)
+}
+
+// genericArnTemplate builds a templated string for constructing a terraform id component which is an ARN of any format.
+// It always includes the aws partition, service, and resource. Unless you specify to elide them, it will also include
+// templates which resolve to the region (from the spec.forProvider) and the account id (calculated from the provider
+// config).
+func genericArnTemplate(service string, resource string, elideAccountId bool, elideRegion bool) string {
```

Comment:
nit: Looks like in all calls to `genericArnTemplate`, the `elideAccountId` is set to `false`. We can eliminate the parameter till we actually need it.